backend/app/views/spree/admin/menus/index.html.erb

```diff
@@ -24,8 +24,8 @@
             </small>
           </td>
           <td>
-            <% menu.store_ids.each do |store| %>
-              <span class="badge badge-secondary"><%= @stores.find(store).unique_name %></span>
+            <% if menu.store_id %>
+              <span class="badge badge-secondary"><%= @stores.find(menu.store_id).unique_name %></span>
```

Comment:
`menu.store.unique_name` should be better for the eyes :)

Reply:
I will sort all of the above, nothing too tricky to fix there.